daemon/snap.go

```diff
@@ -132,7 +132,8 @@ type appJSON struct {
 
 func mapLocal(localSnap *snap.Info, snapst *snapstate.SnapState) map[string]interface{} {
 	status := "installed"
-	if snapst.Active {
+
+	if snapst.Current == localSnap.Revision {
```

Comment:
Because the map being returned talks about a particular revision, rather than the snap as a whole.

I wonder if it's caught by tests, though.